examples/example_app/example_app/views.py

```diff
@@ -77,6 +79,72 @@ class ExampleModelUIViewSet(views.NautobotUIViewSet):
     queryset = ExampleModel.objects.all()
     serializer_class = serializers.ExampleModelSerializer
     table_class = tables.ExampleModelTable
+    object_detail_content = ui.ObjectDetailContent(
+        panels=(
+            ui.ObjectFieldsPanel(
+                section=ui.SectionChoices.LEFT_HALF,
+                weight=100,
+                fields="__all__",
+            ),
+            # A table of objects derived dynamically in `get_extra_context()`
+            ui.ObjectsTablePanel(
+                section=ui.SectionChoices.RIGHT_HALF,
+                weight=100,
+                context_table_key="dynamic_table",
+                max_display_count=3,
+            ),
+            # A table of non-object data with staticly defined columns
+            ui.DataTablePanel(
+                section=ui.SectionChoices.RIGHT_HALF,
+                label="Custom Table 1",
+                weight=200,
+                context_data_key="data_1",
+                columns=["col_1", "col_2", "col_3"],
+                column_headers=["Column 1", "Column 2", "Column 3"],
+            ),
+            # A table of non-object data with dynamic (render-time) columns
+            ui.DataTablePanel(
+                section=ui.SectionChoices.FULL_WIDTH,
+                label="Custom Table 2",
+                weight=100,
+                context_data_key="data_2",
+                context_columns_key="columns_2",
+                context_column_headers_key="column_headers_2",
+            ),
+        ),
+    )
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            # Add dynamic table of objects for custom panel
+            context["dynamic_table"] = CircuitTable(Circuit.objects.restrict(request.user, "view"))
+            # Add non-object data for object detail view custom tables
+            context["data_1"] = [
+                {"col_1": "value_1a", "col_2": "value_2", "col_3": "value_3", "col_4": "not shown"},
+                {"col_1": "value_1b", "col_2": None},
+            ]
+            context["data_2"] = [
+                {
+                    "a": format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=1),
+                    "e": format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=5),
+                    "i": format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=9),
+                    "o": format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=15),
+                    "u": format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=21),
```

Comment:
I'm not sure of the purpose of those urls but I would at least suggest to take the wikipedia address into some variable. Or maybe even go with:
```
{letter: format_html('<a href="https://en.wikipedia.org/wiki/{val}">{val}</a>', val=number) for letter, number in zip('aeiou', [1,5,9,15,21])}
# or if numbers are not needed just iterate over aeiou and put it in the url?
```

Reply:
Purpose was purely just to demonstrate that the data table can contain HTML content if desired/appropriate. I can add some comments to that effect - efficiency of implementation was not a major concern to me since this is just example code.